raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -23,92 +24,173 @@ def comms_nodes(amount_of_nodes) -> {int, CommsNode}:
+    client_1, address_1 = comms_node_1.client, comms_node_1.address
+    client_2, address_2 = comms_node_2.client, comms_node_2.address
 
-    # attempt to locate an unregistered peer
-    assert comms_node.client._get_peer_id(generate_address()) is ""
+    topic_id_1, _ = client_1.subscribe_to(address_2)
+    topic_id_2, _ = client_1.subscribe_to(address_2)
+
+    assert topic_id_1 == topic_id_2
+
+
+@pytest.mark.parametrize("amount_of_nodes", [1])
+def test_is_subscribed_to_invalid(comms_nodes):
+    client, address = comms_nodes[1].client, comms_nodes[1].address
+
+    # check subscription to non-subscribed address
+    assert client.is_subscribed_to(address) is False
+
+    # check subscription to unregistered address
+    assert client.is_subscribed_to(generate_address()) is False
 
 
-# FIXME: comms node prints strange ServerUnaryCall message
 @pytest.mark.parametrize("amount_of_nodes", [1])
-def test_has_subscriber_self(comms_nodes):
+def test_is_subscribed_to_self(comms_nodes):
     comms_node = comms_nodes[1]
     client, address = comms_node.client, comms_node.address
 
+    # no subscription should be present
+    assert client.is_subscribed_to(address) is False
+
     client.subscribe_to(address)
 
     # subscribe to self and check subscription
     assert client.is_subscribed_to(address) is True
 
 
 @pytest.mark.parametrize("amount_of_nodes", [2])
-def test_has_subscriber(comms_nodes):
+def test_is_subscribed_to_peers(comms_nodes):
     comms_node_1, comms_node_2 = comms_nodes[1], comms_nodes[2]
     client_1, address_1 = comms_node_1.client, comms_node_1.address
     client_2, address_2 = comms_node_2.client, comms_node_2.address
 
-    # FIXME: nodes shouldn't have to subscribe to themselves for this to work
-    client_1.subscribe_to(address_1)
-    client_2.subscribe_to(address_2)
+    # no subscriptions should be present
+    assert client_1.is_subscribed_to(address_1) is False
+    assert client_1.is_subscribed_to(address_2) is False
+    assert client_2.is_subscribed_to(address_1) is False
+    assert client_2.is_subscribed_to(address_2) is False
 
     # subscribe from node 1 to 2
     client_1.subscribe_to(address_2)
 
     # check subscriptions
-    assert client_1.is_subscribed_to(address_1) is True
+    assert client_1.is_subscribed_to(address_1) is False
     assert client_1.is_subscribed_to(address_2) is True
     assert client_2.is_subscribed_to(address_1) is False
-    assert client_2.is_subscribed_to(address_2) is True
+    assert client_2.is_subscribed_to(address_2) is False
+
+    # now from node 2 to 1 and check again
+    client_2.subscribe_to(address_1)
+
+    assert client_1.is_subscribed_to(address_1) is False
+    assert client_1.is_subscribed_to(address_2) is True
+    assert client_2.is_subscribed_to(address_1) is True
+    assert client_2.is_subscribed_to(address_2) is False
+
+
+@pytest.mark.parametrize("amount_of_nodes", [1])
+@pytest.mark.skip(reason="hangs, incomplete")
```

Comment:
added to the pending issues doc.

it's not a huge deal for us. but it's worrying because it seems to cause problems in the caller as well as the comms node

Reply:
I think this is a big deal for us. 

If a partner of a node goes offline (and the rif comms node related to it too), the behaviour will be that the sender hangs..

Reply:
> I think this is a big deal for us.
> 
> If a partner of a node goes offline (and the rif comms node related to it too), the behaviour will be that the sender hangs..

not sure if this is the case, though. an unregistered address is not the same as an offline node.

i've added a note to [RLP-956](https://jirainfuy.atlassian.net/browse/RLP-956) so that we can have an integration test for this. we'll see what happens then

Reply:
an offline node will not be part of the dht after 6 hrs

a unregistered address will not be part too